com.reprezen.swagedit.tests/src/com/reprezen/swagedit/editor/PathParamHyperlinkDetectorTest.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2016 ModelSolv, Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ModelSolv, Inc. - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package com.reprezen.swagedit.editor;
+
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.text.hyperlink.IHyperlink;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PathParamHyperlinkDetectorTest {
+
+	private PathParamHyperlinkDetector detector = new PathParamHyperlinkDetector();
+	private ITextViewer viewer;
+
+	@Before
+	public void setUp() {
+		viewer = mock(ITextViewer.class);
+	}
+
+	@Test
+	public void testShouldCreateHyperLink_FromPathParameter_ToParameterDefinition() throws Exception {
+		SwaggerDocument document = new SwaggerDocument();
+		when(viewer.getDocument()).thenReturn(document);
+
+		String text = 
+				"paths:\n" +
+				"  /{id}:\n" +
+				"    get:\n" +
+				"      parameters:\n" +
+				"        - name: id\n" +
+				"          type: number\n"+
+				"          required: true\n" +
+				"          in: path\n";
+
+		document.set(text);
+		// region that includes `/{id}:`
+		IRegion region = new Region("paths:\n  /{i".length(), 1);
+		IHyperlink[] hyperlinks = detector.detectHyperlinks(viewer, region, false);
+
+		// expected region
+		IRegion linkRegion = new Region(document.getLineOffset(1) + "  /".length(), "{id}".length());
+		IRegion targetRegion = new Region(document.getLineOffset(4), 86);
```

Comment:
I like the approach with mocks you found to test `HyperlinkDetector`. Testing editors is hard, I am glad that you found a relatively easy way to test this.